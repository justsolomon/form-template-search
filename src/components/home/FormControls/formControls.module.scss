@import 'styles/variables';
@import 'styles/mixins';

.form-controls {
  justify-content: space-between;

  &,
  &__filters,
  &__filters__selects {
    @include flex-center();
  }

  &__filters {
    &__selects {
      > div {
        width: 160px;
        margin-left: 20px;
      }
    }

    &__label {
      color: $secondary-color;
    }
  }

  @media #{$media-desktop--sm} {
    flex-direction: column;
    align-items: flex-start;

    &__filters {
      margin-top: 24px;
    }
  }

  @media #{$media-tablet} {
    &__filters__selects > div {
      width: 150px;
    }
  }

  @media #{$media-mobile} {
    &__search {
      width: 100%;
    }

    &__filters {
      width: 100%;
      flex-direction: column;
      align-items: flex-start;

      &__label {
        margin-bottom: 18px;
      }

      &__selects {
        width: 100%;
        justify-content: space-between;

        > div {
          margin: 0;
        }
      }
    }
  }

  @media #{$media-mobile--sm} {
    &__filters__selects {
      flex-direction: column;

      > div {
        width: 100%;
        margin-bottom: 16px;
      }
    }
  }
}
