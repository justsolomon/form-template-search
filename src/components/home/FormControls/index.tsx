import Input from 'components/global/Input';
import Select from 'components/global/Select';
import { ReactComponent as SearchIcon } from 'assets/vectors/search.svg';
import { TemplateCategory, TemplateOrder } from 'types/global';
import { useAppDispatch, useAppSelector } from 'redux/hooks';
import {
  resetPageState,
  search,
  sortByAlphabet,
  sortByDate,
  updateCategory,
} from 'redux/slices/templateSlice';
import styles from './formControls.module.scss';

const orderOptions: TemplateOrder[] = ['Default', 'Ascending', 'Descending'];

const FormControls = () => {
  const { searchQuery, filters } = useAppSelector((store) => store.template);
  const dispatch = useAppDispatch();

  return (
    <div className={styles['form-controls']}>
      <Input
        placeholder="Search Templates"
        rightIcon={<SearchIcon />}
        value={searchQuery}
        onChange={(e) => {
          dispatch(search(e.target.value));
          dispatch(resetPageState());
        }}
        containerClass={styles['form-controls__search']}
      />

      <div className={styles['form-controls__filters']}>
        <p className={styles['form-controls__filters__label']}>Sort By:</p>
        <div className={styles['form-controls__filters__selects']}>
          <Select
            label="Category"
            options={['All', 'Health', 'E-commerce', 'Education']}
            value={filters.category}
            onChange={(e) => {
              dispatch(updateCategory(e.target.value as TemplateCategory));
              dispatch(resetPageState());
            }}
          />
          <Select
            label="Order"
            options={orderOptions}
            value={filters.alphabetical}
            onChange={(e) => {
              dispatch(sortByAlphabet(e.target.value as TemplateOrder));
              dispatch(resetPageState());
            }}
          />
          <Select
            label="Date"
            options={orderOptions}
            value={filters.date}
            onChange={(e) => {
              dispatch(sortByDate(e.target.value as TemplateOrder));
              dispatch(resetPageState());
            }}
          />
        </div>
      </div>
    </div>
  );
};

export default FormControls;
