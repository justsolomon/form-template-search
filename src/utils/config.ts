export const baseUrl =
  'https://front-end-task-dot-result-analytics-dot-fpls-dev.uc.r.appspot.com/api/v1';
