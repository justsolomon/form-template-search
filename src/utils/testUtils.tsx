import {
  queries,
  Queries,
  render as rtlRender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import rootReducer from 'redux/rootReducer';
import { Store } from 'redux';

interface CustomRenderOptions extends RenderOptions {
  preloadedState: Record<string, any>;
  store: Store;
}

const render = <
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
>(
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  }: CustomRenderOptions,
): RenderResult => {
  function Wrapper({ children }: { children: React.ReactNode }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, {
    wrapper: Wrapper as React.ComponentType,
    ...renderOptions,
  });
};

export * from '@testing-library/react';
export { render };
