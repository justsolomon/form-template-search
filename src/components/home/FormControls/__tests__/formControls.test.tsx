import { render } from 'utils/testUtils';
import store from 'redux/store';
import FormControls from '..';

describe('<FormControls />', () => {
  it('should render the FormControls component correctly', () => {
    const { asFragment } = render(<FormControls />, {
      preloadedState: {},
      store,
    });

    expect(asFragment).toMatchSnapshot();
  });
});
