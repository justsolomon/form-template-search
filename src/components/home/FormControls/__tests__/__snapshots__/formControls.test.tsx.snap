// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormControls /> should render the FormControls component correctly 1`] = `[Function]`;
